File: exercise/C#/day25/TimeCapsule/Browser.cs
using System.Diagnostics;

namespace TimeCapsule;

public static class Browser
{
    public static void Open(string filePath)
        => new Process
        {
            StartInfo = new ProcessStartInfo
            {
                FileName = filePath,
                UseShellExecute = true
            }
        }.Start();
}

File: exercise/php/day25/src/TimeCapsule/Browser.php
<?php

namespace TimeCapsule;

class Browser
{
    public static function open(string $filePath): void
    {
        if (PHP_OS_FAMILY === 'Windows') {
            shell_exec('start ' . escapeshellarg($filePath));
        } elseif (PHP_OS_FAMILY === 'Darwin') {
            shell_exec('open ' . escapeshellarg($filePath));
        } else {
            shell_exec('xdg-open ' . escapeshellarg($filePath));
        }
    }
}